packages/synapse-react-client/src/components/DataGrid/StartGridSession.tsx

```diff
@@ -125,23 +129,31 @@ export const StartGridSession = ({
     }
   }
 
+  useEffect(() => {
+    if (!allowInput && query) {
+      handleStartSession(query)
+    }
+  }, [query])
+
   return (
```

Comment:
Could we simplify to something like this? Have a simple return here if `!allowInput` and remove the `allowInput` checks below
```suggestion
  if (!allowInput) { 
    return null
  }
  return (
```